### PR TITLE
Feature :: Duplicate column step

### PR DESCRIPTION
**Pull request by vdestraitt**
Closes https://github.com/ToucanToco/vue-query-builder/issues/213

### CODE REVIEW

File: src/lib/steps.ts [hunk 1]
```diff
@@ -48,6 +48,11 @@ export type DomainStep = {
   domain: string;
 };
 
+export type DuplicateColumnStep = {
+  name: 'duplicate';
+  column: string;
```
Comment (adimascio) [inline src/lib/steps.ts]: Shouldn't we, for consistency-sake with other steps, add a `new_column?: string;` field?

File: src/components/stepforms/DuplicateColumnStepForm.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  @Prop({ type: Object, default: () => ({ name: 'duplicate', column: '' }) })
+  initialStepValue!: DuplicateColumnStep;
+
+  readonly title: string = 'Duplicate Column Step';
+
+  // Only manage the deletion of 1 column at once at this stage
+  editedStepModel = duplicateSchema;
+  column: string = this.initialStepValue.column;
+
+  stepToValidate() {
```
Comment (adimascio) [inline src/components/stepforms/DuplicateColumnStepForm.vue]: Why is this needed? It looks like the default behaviour to me.

File: src/components/stepforms/DuplicateColumnStepForm.vue [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  set stepSelectedColumn(colname: string | null) {
+    if (colname === null) {
+      throw new Error('should not try to set null on duplicate "column" field');
+    }
+    if (colname !== null) {
+      this.column = colname;
+    }
+  }
+
+  rebuildStep() {
```
Comment (adimascio) [inline src/components/stepforms/DuplicateColumnStepForm.vue]: Same as above, Is this needed at all?

Comment (adimascio) [inline src/components/stepforms/DuplicateColumnStepForm.vue]: Shouldn't we check that we're not overriding an existing column in the dataset?

File: src/lib/steps.ts (hunk 1 above)
Comment (vdestraitt) [reply src/lib/steps.ts]: @adimasci it was on purpose: we wanted, as for the `delete` step, to get the column creation immediately effective after clicking, without having to fill in any form field. It is inspired from other softwares and looks lighter, offering a more direct feedback to the user.

But after reflexion, I think that you're right and that in our context we should ask for the new name in a form at creation, because a duplicated column will need a rename in 95% of the cases.
And it would be way heavier than other softwares I had in mind if we have to log an additional `rename` step in the pipeline every time we duplicate a column !

So I will amend this ! Thank you
